simplecpp.cpp

```diff
@@ -3123,6 +3123,24 @@ static std::string openHeader(std::ifstream &f, const simplecpp::DUI &dui, const
     return ret;
 }
 
+static bool IsFileExists(const std::string& simplePath)
+{
+	if (simplePath.empty())
+	{
+		return false;
+	}
+
+	// This is a workaround to properly check if a file exists,
+	// since std::ifstream::good() incorrectly returns true for directories
+	std::fstream fs(simplePath, std::ios::app);
```

Comment:
That is an interesting handling.

Reply:
This solution still has flaws; when the file does not exist, an empty file will be created. Therefore, I replaced it with
```
std::fstream fs(simplePath, std::ios::in|std::ios::out);
```

Here is a simple test result (linux, mac, and windows):
https://github.com/foryoung365/CppFstreamTest/actions/runs/11549258974

Here are some references:
https://en.cppreference.com/w/cpp/io/basic_filebuf/open